fission/src/ui/panels/configuring/ConfigureJointsPanel.tsx

```diff
@@ -0,0 +1,259 @@
+import Label, { LabelSize } from "@/ui/components/Label"
+import Panel, { PanelPropsImpl } from "@/ui/components/Panel"
+import ScrollView from "@/ui/components/ScrollView"
+import Slider from "@/ui/components/Slider"
+import Stack, { StackDirection } from "@/ui/components/Stack"
+import { Box } from "@mui/material"
+import { useMemo, useState } from "react"
+import { FaGear } from "react-icons/fa6"
+
+type JointRowProps = {
+    robot: MirabufSceneObject
+    driver: Driver
+}
+
+const JointRow: React.FC<JointRowProps> = ({ robot, driver }) => {
+    const driverSwitch = (driver: Driver, slider: unknown, hinge: unknown, drivetrain: unknown) => {
+        switch (driver.constructor) {
+            case SliderDriver:
+                return slider
+            case HingeDriver:
+                return hinge
+            case WheelDriver:
+                return drivetrain
+            default:
+                return drivetrain
+        }
+    }
+
+    const [velocity, setVelocity] = useState<number>(
+        ((driver as SliderDriver) || (driver as HingeDriver) || (driver as WheelDriver)).maxVelocity
+    )
+    const [force, setForce] = useState<number>(
+        ((driver as SliderDriver) || (driver as HingeDriver) || (driver as WheelDriver)).maxForce
+    )
+
+    const onChange = (vel: number, force: number) => {
+        if (driver instanceof WheelDriver) {
+            const wheelDrivers = robot?.mechanism
+                ? World.SimulationSystem.GetSimulationLayer(robot.mechanism)?.drivers.filter(
+                      x => x instanceof WheelDriver
+                  )
+                : undefined
+            wheelDrivers?.forEach(x => {
+                x.maxVelocity = vel
+                x.maxForce = force
+            })
+
+            // Preferences
+            PreferencesSystem.getRobotPreferences(robot.assemblyName).driveVelocity = vel
+            PreferencesSystem.getRobotPreferences(robot.assemblyName).driveAcceleration = force
+        } else {
+            ((driver as SliderDriver) || (driver as HingeDriver)).maxVelocity = vel
+            ;((driver as SliderDriver) || (driver as HingeDriver)).maxForce = force
+
+            // Preferences
+            if (driver.info && driver.info.name) {
+                const removedMotor = PreferencesSystem.getRobotPreferences(robot.assemblyName).motors
+                    ? PreferencesSystem.getRobotPreferences(robot.assemblyName).motors.filter(x => {
+                          if (x.name) return x.name != driver.info?.name
+                          return false
+                      })
+                    : []
+
+                removedMotor.push({
+                    name: driver.info?.name ?? "",
+                    maxVelocity: vel,
+                    maxForce: force,
+                })
+
+                PreferencesSystem.getRobotPreferences(robot.assemblyName).motors = removedMotor
+            }
+        }
+
+        PreferencesSystem.savePreferences()
+    }
```

Comment:
I believe in react it's better practice to use a useCallback hook in cases like this, but @PepperLola or @HunterBarclay could you confirm? It would look something like this:

```ts
const onChange = useCallback((vel: number, force: number) => {
    // Function body
}, [driver, velocity, force]);
```